crates/grafbase-sdk/src/types/selection_set.rs

```diff
@@ -134,6 +134,14 @@ impl<'a> SelectionSet<'a> {
         self.fields_ordered_by_parent_entity()
     }
 
+    /// Returns the field at the given index in this selection set.
+    pub fn field(&self, index: usize) -> Option<Field<'a>> {
```

Comment:
Feels like a bit weird API to me, either you need the position with `enumerate()` or you don't and `Field` is `Copy`. So curious to see how you use it, but nothing against it either 🤷